whitepapers/tickets/SHARE-822.md

```diff
@@ -0,0 +1,112 @@
+| 1   | Bepress       | False   |
+| 1   | My custom tax | False   |
+
+## Subject
+
+| Column        |   Type   | Indexed | Nullable | FK  | Default | Description                                                                                        |
+| :------------ | :------: | :-----: | :------: | :-: | :-----: | :------------------------------------------------------------------------------------------------- |
+| name          |   text   |         |          |     |         | The subject name                                                                                   |
+| deleted       |   bool   |         |          |     |         |                                                                                                    |
+| handle        |   text   | unique  |          |     |         | A unique refence to this subject that Sources may use to specify which subject needs to be altered |
+| taxonomy_id   |   int    |    ✓    |          |  ✓  |         | The taxonomy this subject belongs to                                                               |
+| alias_id      |   int    |    ✓    |          |  ✓  |         | The Bepress equivalent of this subject. Set to id if this subject belongs to Bepress.              |
+| parent_id     |   int    |    ✓    |    ✓     |  ✓  |         |                                                                                                    |
+| date_created  | datetime |         |          |     |         |                                                                                                    |
+| date_modified | datetime |  Desc   |          |     |         |                                                                                                    |
+
+| id  | name       | deleted | handle    | taxonomy_id | bepress_id | parent_id |
+| --- | ---------- | ------- | --------- | ----------- | ---------- | --------- |
+| 1   | Law        | False   | Law       | 1           | 1          | NULL      |
+| 2   | Civil Law  | False   | Civil Law | 1           | 2          | 1         |
+| 3   | Custom Law | False   | 538df3e1  | 2           | 1          | 1         |
+
+## Changes to OSF: 
+  - Whenever a custom taxonomy if modified the ne
+  - The on_preprint updated method will now serialize subjects as:
+
+## Situations:
+
+### Reference
+Taxonomy X:
+  Subject Alpha -> Bepress A
+    Subject Bravo -> Bepress B
+      Subject Charlie -> Bepress C
+    Subject Delta -> Bepress B
+  Subject Echo -> Bepress D
+
+Given SHARE is aware of custom Taxonomy X
+When SHARE recieves a work, "Gamma", with subjects [Alpha, Bravo, Charlie]
+Then SHARE will remove all parent subjects as they will be implied by subject "Charlie"
+
+  Reasoning:
+    If at any point "Charlie" is moved to another location in Taxonomy X no changes need to
+    be made to "Gamma", we simply find all works that haven't been modified since "Charlie"
+    has been modified and re-index them.
```

Comment:
I guess that makes sense. The only weird bit might be when nothing is actually changed on the source's side (e.g. there was an error in the taxonomy we were given), but we require them to mark it updated so we'll reharvest it. For now, I think that's fine. But if/when it's super easy for us to flag a SUID for reingestion and have a janitor or something pick it up, that seems like the ideal solution.